### PR TITLE
doc: Manual + Fixes

### PR DESCRIPTION
**Pull request by johannes-wolf**
This updates the manual:
- Add a title page
- Change the TOC to a two column layout (we have quite short headings, so there was a lot of wasted horizontal space)
- Improve examples (#296)
- Add group anchor explanation

This also fixes two bugs in `coordinates.typ`:
- We should allow zero distance lerp, i.e. `(a, .5, a)` without failing (div. by zero) but returnung `a`. This is useful when doing calculations depending on anchors.
- The coordinate systems "|-" and "-|" must not update the current coordinate to `a` or `b` when resolving both sides.

### CODE REVIEW

**Comment by johannes-wolf** [comment]
@fenjalien What do you think? I want to have a nicer looking title page for the manual.

**Review by fenjalien — CHANGES_REQUESTED** [review]
Looks cool! Do you think we could show some examples/demos on the title? It would be even better to come up with an icon, something like typst guy, but thats much harder.

Also either you have more blockers on this PR or you've messed up your history as you've got changes to charts in here...

File: manual.typ [hunk 1]
```diff
@@ -87,27 +87,57 @@
...
+    content((), text(left-color, 3cm)[The CeTZ\ Package],
+      name: "title")
+    content((rel: (0,-2), to: "title.south-west"),
+      text(20pt, authors.map(v => link(v.at(1), [#v.at(0)])).join("\n")),
+      anchor: "north-west", name: "subtitle")
+    content((rel: (0,-2), to: "subtitle.south-west"),
+      text(20pt, link(url, [Version ] + lib.version.map(v => [#v]).join("."))),
+      anchor: "north-west", name: "version")
+  })
+  pagebreak(weak: true)
```
[the remark this answers is not in the document]
Comment (johannes-wolf) [reply manual.typ:121]: History is fucked up. I will rebase after the anchor branch got merged.

**Comment by johannes-wolf** [comment]
> Looks cool! Do you think we could show some examples/demos on the title? It would be even better to come up with an icon, something like typst guy, but thats much harder.
> 
> Also either you have more blockers on this PR or you've messed up your history as you've got changes to charts in here...

Yes, examples would be nice. But what should we show? @JamesxX, any ideas?

File: manual.typ [hunk 2]
```diff
@@ -146,20 +176,30 @@ stroke(none)
...
+        content(
+          (rel: ("group.center", .75, "group." + n),
+           to: "group." + n), n)
+      } else {
+        content((rel: (0, .5), to: "group.center"), n)
+      }
+      circle("group." + n, radius: .1, fill: black)
+    })
+  })
+})
```
[the remark this answers is not in the document]
Comment (johannes-wolf) [reply manual.typ:197]: Yes, we should add this for all elements. I just wanted to have at least one example image.

**Review by fenjalien — APPROVED** [review]

**Pull request merged**